doc/format_specifications.rdoc

```diff
@@ -163,34 +180,126 @@ Format +argument+ as a binary integer:
 
   sprintf('%b', 1)  # => "1"
   sprintf('%b', 4)  # => "100"
+
   # Prefix '..' for negative value.
   sprintf('%b', -4) # => "..100"
 
-Type-specific modifier:
+  # Alternate format.
+  sprintf('%#b', 4)  # => "0b100"
+  sprintf('%#B', 4)  # => "0B100"
 
-- '#' flag' (use alternate format):
+==== Specifier +c+
 
-    sprintf('%#b', 1)  # => "0b1"
-    sprintf('%#B', 1)  # => "0B1"
-    sprintf('%#b', 4)  # => "0b100"
+Format +argument+ as a single character:
 
-==== Specifier +c+
+  sprintf('%c', 'A') # => "A"
+  sprintf('%c', 65)  # => "A"
+
+Flags do not apply.
 
 ==== Specifier +d+
 
+Format +argument+ as a decimal integer:
+
+  sprintf('%d', 100)  # => "100"
+  sprintf('%d', -100) # => "-100"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +e+ and +E+
 
+The two specifiers +e+ and +E+ behave identically
+except when flag <tt>'#'</tt>+ is used.
+
+Format +argument+ in
+{scientific notation}[https://en.wikipedia.org/wiki/Scientific_notation]:
+
+  sprintf('%e', 3.14159)  # => "3.141590e+00"
+  sprintf('%e', -3.14159)  # => "-3.141590e+00"
+  sprintf('%#e', 3.14159)  # => "3.141590e+00"
+  sprintf('%#E', 3.14159)  # => "3.141590E+00"
+
 ==== Specifier +f+
 
+Format +argument+ as a floating-point number:
+
+  sprintf('%f', 3.14159)  # => "3.141590"
+  sprintf('%f', -3.14159) # => "-3.141590"
+
+Flag <tt>'#'</tt> does not apply.
+
 ==== Specifiers +g+ and +G+
 
+Format +argument+ in a "general" format that may be convenient
+for human readers:
```

Comment:
I don't think this accurately describes what this specifier does.

It seems to work the same as C's `g` and `G` specifiers.

```suggestion
Format +argument+ using exponential form (+e+/+E+ specifier)
if the exponent is less than -4 or greater than or equal to the precision.
Otherwise format +argument+ using floating-point form +f+ specifier).

The precision defaults to 6, and can be changed to +n+ using the +.n+ flag.
```

We should also have some examples involving changing the precision. For example:

```ruby
sprintf('%g', 100000000000) #=> "1e+11"
sprintf('%.12g', 100000000000) #=> "100000000000"
```